ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+
+# Copyright 2014-2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import os
+import re
+import subprocess
+import sys
+import time
+
+from xml.sax.saxutils import escape
+from xml.sax.saxutils import quoteattr
+
+import yaml
+
+
+def main(argv=sys.argv[1:]):
+    config_file = os.path.join(
+        os.path.dirname(__file__), 'configuration', '.clang-tidy')
+    extensions = ['c', 'cc', 'cpp', 'cxx', 'h', 'hh', 'hpp', 'hxx']
+
+    parser = argparse.ArgumentParser(
+        description='Check code style using clang_tidy.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--config',
+        metavar='path',
+        default=config_file,
+        dest='config_file',
+        help='The config file')
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help='The files or directories to check. For directories files ending '
+             'in %s will be considered.' %
+             ', '.join(["'.%s'" % e for e in extensions]))
+
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    parser.add_argument(
+        '--export-fixes',
+        help='Generate a DAT file of recorded fixes')
+    parser.add_argument(
+        '--explain-config',
+        action='store_true',
+        help='Explain the enabled checks')
+    parser.add_argument(
+        '--fix-errors',
+        action='store_true',
+        help='Fix the suggested changes')
+    args = parser.parse_args(argv)
+
+    if not os.path.exists(args.config_file):
+        print("Could not find config file '%s'" % args.config_file,
+              file=sys.stderr)
+        return 1
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    files = get_files(args.paths, extensions)
+    if not files:
+        print('No files found', file=sys.stderr)
+        return 1
+
+    bin_names = [
+        'clang-tidy',
+        'clang-tidy-6.0',
```

Comment:
You may want to add `clang-tidy-3.8` here, as that is the version on Xenial.

Reply:
Why is the version from Xenial relevant here? Our current target platform is Bionic which ships 6.0.

Reply:
For ROS 2, yes.  For another project that we are considering using this on, they still target Xenial.  The other linters also generally support older versions, so it seems like we should follow that trend.

Reply:
I would expect significant differences between version 6.0 and 3.8. So I am not sure if supporting such wide spread is feasible.